### ISSUE TITLE
Fonts and Images

### ISSUE THREAD

**Issue by tommykent1210** [jasonlewis/basset#194]
I don't know if this has bee answered before, but is there any way to make Basset handle fonts too?

The issue is, I use Fontawesome and Laravel. Basset obviously compiles all CSS to the builds folder in one nice css file. However, Fontawesome's Font files are in /components/fontawesome/font and the fontawesome css file references them relatively (via: 
  src: url('../font/fontawesome-webfont.eot?v=3.2.1'); )

This presents a problem, because the fonts aren't in /builds/font.

Now, I could alter the CSS files at source to reference them absolutely BUT my application is free to distribute, so site owners use composer to bring down the fontawesome files.

So, unless basset has a way to manage fonts I'm a little stuck.

Thanks
Tom

**Comment by Anahkiasen** [comment]
Well there is the `UriRewriteFilter` filter but I think it's broken on develop.

**Comment by tommykent1210** [comment]
Broken on develop? How about master?

Also, is there any documentation on the UriRewriteFilter? I've never really used the filters before.

I've looked at: http://jasonlewis.me/code/basset/4.0/filters

but an example would be much appreciated.

Thanks

**Comment by jasonlewis** [comment]
Is font awesome inside or outside of the `public` directory?

**Comment by tommykent1210** [comment]
Fontawesome is located in:

ROOT_FOLDER/components/font-awesome/

Basset puts the builds in:
ROOT_FOLDER/public/builds/....

Thanks

**Comment by jasonlewis** [comment]
Yeah that probably won't work too well. They're not web accessible.

**Comment by tommykent1210** [comment]
hmmmm, This presents a real problem then. 

My app is meant to be nice and easy to install, clone the git repo (or download the zip and upload it) the install compose and "composer install". All the configs are in the git repo so I was hoping for everything to work out of the box. I don't believe there are any easy ways to move these without manually requiring users to do it.

Any suggestions? :/

**Comment by jasonlewis** [comment]
Why can't you put all your assets in public?

**Comment by tommykent1210** [comment]
I could, but I'd have to move them manually. Font-awesome is installed via composer and composer puts them in /components as opposed to /public. Unless theres a way to change that?

**Comment by Anahkiasen** [comment]
On the other hand, you should _not_ manage your front-end assets with Composer. That's not what it was meant for nor what it's useful for. I recommend you switch to an actual assets manager like [Bower](http://bower.io/) and create the following files at the root of your repository :

`.bowerrc`

``` json
{
  "directory" : "public/components"
}
```

`bower.json`

``` json
{
  "dependencies": {
    "font-awesome": "latest"
  }
}
```

**Comment by jasonlewis** [comment]
What Maxime said. Composer a PHP dependency manager. Use Bower, it's really
good.

**Comment by petercoles** [comment]
Or if you want to stick with Composer, update your composer.json config setting to override the default location for components, e.g.

```
    "config": {
        "preferred-install": "dist",
        "component-dir": "public/components"
    },
```

**Comment by eviweb** [comment]
Hi,

I am just new to Laravel and then to Basset which seems to me a great tool, thanks Jason ;)
Going on my understanding of both products, I ran into the same issue than tommykent1210.
After having installed semantic-ui via composer, configured a collection using $collection->requireTree() method and filtered paths with UriRewriteFilter, I relied on fonts and images to be part of the collection build. But after a quick check, I realized that only js and css files have been copied to /public/build/ directory.
Aren't images and fonts considered as assets ?
IMHO, they should be included in a collection build, like js and css files are, don't you think so ?

Eric

**Comment by MonsterGfx** [comment]
For what it's worth, I solved the problem by adjusting the path in my .less file. Here's my directory tree:

```
public/
    fonts/   (containing my font-awesome font files)
    assets/    (as recommended by the basset docs)
    builds/    (as recommended by the basset docs)
```

My font-awesome variables.less file now contains the following line:

```
@FontAwesomePath:    "../../../../../../../font";
```

So that it points from the font-awesome .css file deep in the public/builds/... tree to the fonts directory in the root folder.

Not ideal, but it works. A better solution would be the one eviweb suggested: copy arbitrary assets like fonts/images/etc. to the build directory.
